Index: include/ISceneNode.h
===================================================================
--- include/ISceneNode.h	(revision 4020)
+++ include/ISceneNode.h	(working copy)
@@ -55,6 +55,7 @@
 				parent->addChild(this);
 
 			updateAbsolutePosition();
+			UserData = NULL;
 		}
 
 
@@ -737,6 +738,20 @@
 		/** \return The node's scene manager. */
 		virtual ISceneManager* getSceneManager(void) const { return SceneManager; }
 
+		//! Set User Data
+		/** \param data The data to set it to. */
+		inline virtual void setUserData(void *data)
+		{
+			UserData = data;
+		}
+
+		//! Get User Data
+		/** \return The data. */
+		inline virtual void *getUserData()
+		{
+			return UserData;
+		}
+
 	protected:
 
 		//! A clone function for the ISceneNode members.
@@ -838,6 +853,9 @@
 
 		//! Is debug object?
 		bool IsDebugObject;
+
+		//! User data
+		void* UserData;
 	};
 
 
